src/RealtimeClient.ts

```diff
@@ -452,11 +467,28 @@ export default class RealtimeClient {
     this.log('transport', `connected to ${this._endPointURL()}`)
     this._flushSendBuffer()
     this.reconnectTimer.reset()
-    this.heartbeatTimer && clearInterval(this.heartbeatTimer)
-    this.heartbeatTimer = setInterval(
-      () => this._sendHeartbeat(),
-      this.heartbeatIntervalMs
-    )
+    if (!this.worker) {
+      this.heartbeatTimer && clearInterval(this.heartbeatTimer)
+      this.heartbeatTimer = setInterval(
+        () => this._sendHeartbeat(),
+        this.heartbeatIntervalMs
+      )
+    } else {
+      this.log('worker', `starting worker for from ${this.workerUrl!}`)
+
+      this.workerRef = new Worker(this.workerUrl!)
+      this.workerRef.onerror = (error) => {
+        this.log('worker', 'worker error', error.message)
+        this.workerRef!.terminate()
+      }
```

Comment:
we'll need retry logic here to start up another worker instance.

Reply:
if this does not happen is due to script not found or not able to create the worker at all meaning that retrying won't do much 😞